### ISSUE TITLE
Filter Accounts by regular expressions

### ISSUE THREAD

**Issue by SirVer** [aumayr/fava#342]
tl;dr: Filtering for many accounts in the account dropdown would be very useful.

Maybe I am doing it wrong, but this is my use case right now: I want to understand if a building loan agreement was a net win or loss for me. I have a couple accounts around this contract: `Assets:Company:Account`, `Expenses:Company:Taxes`, `Expenses:Company:Comissions` and `Income:Company:StateSupport`. 

Every year I transfer some money from one of my accounts into `Assets:Company:Account` and I get some state supported money into the same account. At the end of the year I need to pay a commission and capital gain taxes go away. I am right now interested in the net gain/losses. 

I wanted to filter on these accounts in Fava - the way I achieved this now is by
using tags - each transaction related to this contract gets the same tag. This gives me exactly the reporting I wanted, but is little cumbersome since the transactions are spread over my beancount file and I have to remember to add the tag should I add new ones.

In bean query I can use something along the lines of `SELECT * WHERE account ~ ".*:Company:.*"` to get the same transactions. It would be cool if I could use this as a filter in the Account drop down menu. Focusing on one single account like it is supported right now is not too useful to me. 

Sorry for the wall of text - I am interested how you handle similar use cases.

**Comment by aumayr** [comment]
The "Account" filter currently only supports selecting one account (not multiple ones, like the "Tags" filter).

Other than that, only two ways come to mind (that you already discovered): Using `bean query` (you can do this within Fava, and even export the results as CSV/Excel/etc.), and using tags. 

I'll mark this as a "feature" and will have a look if implementing filtering for multiple accounts is possible. 

Anyhow, thanks for asking/reporting! :)

**Comment by yagebu** [comment]
Thanks for the suggestion, that's a nice enhancement! The example you've given should work now (and is also the example in the help pages ;) )

Now both a regular expression match and the account name match are done, so if you're not using regular expressions, everything will work as before.

PS: Thanks for UltiSnips!

**Comment by SirVer** [comment]
Thank you for adding this feature! fava is bean-count's killer feature for me, I am immensely grateful for the work you put into it. 

> PS: Thanks for UltiSnips!

You are very welcome :). I like the thought that UltiSnips helped making fava. Btw, are you also using Vim for editing beancount files? I try to get account autocompletion to work, but no cigar so far.

**Comment by yagebu** [comment]
> Btw, are you also using Vim for editing beancount files? I try to get account autocompletion to work, but no cigar so far.

Yes, I don't have autocompletion though, only standard completion which I can access with `Ctrl+Space`. It's all in [yagebu/dotfiles](https://github.com/yagebu/dotfiles) if you're interested, but I guess you also got that far.

**Comment by SirVer** [comment]
I looked through your dotfiles. Seems like you are using the default plugin from https://github.com/nathangrigg/vim-beancount. That does not work for me: For some reason YCM gets really slow with that for me and it also does not suggest my accounts for completion - the omnifunc always returns an empty list. I was so far unable to track down the problem - but I suspect it is because my beancount files are spread over many files and only 'included' in the top level one.
